src/main/resources/assets/computercraft/lua/rom/programs/monitor.lua

```diff
@@ -39,7 +39,40 @@ local function resume( ... )
     return param
 end
 
-local ok, param = pcall( function()
+local timers = { n = 0 }
+
+-- add a timer to the timers list with id 'id', xpos 'x', ypos 'y'
+local function addTimer( id, x, y )
+    timers.n = timers.n + 1
+    timers[timers.n] = { n = 3, id, x, y }
+end
+
+-- remove a timer from the timers list with id 'id'
+local function removeTimer( id )
+    for i = 1, timers.n do
+        if timers[i][1] == id then
+            local x, y = table.unpack( timers[i], 2, timers[i].n )
+            table.remove( timers, i )
+            timers.n = timers.n - 1
+            return true, x, y
+        end
+    end
+    return false
+end
+
+local ok, param = pcall( parallel.waitForAny,
+function()
+    while true do
+        local _, id = os.pullEventRaw( "timer" )
+        -- when a timer is set off, check if it exists and remove it if so
+        -- also generate "mouse_up" event
+        local ok, x, y = removeTimer( id )
+            if ok then
```

Comment:
Why's the indentation here all screwy?

Reply:
attempting to match indentation that squid has here (4 spaces), must have made a mistake while adding more spaces.